tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -2004,8 +2004,10 @@ SpirvInstruction *SpirvEmitter::processCall(const CallExpr *callExpr) {
       // getObject().objectMethod();
       // Also, any parameter passed to the member function must be of Function
       // storage class.
-      needsTempVar = objInstr->isRValue() ||
-                     objInstr->getStorageClass() != spv::StorageClass::Function;
+      needsTempVar =
+          objInstr->isRValue() ||
+          (objInstr->getStorageClass() != spv::StorageClass::Function &&
+           objInstr->getStorageClass() != spv::StorageClass::Workgroup);
```

Comment:
What about variables in the `Private` storage class? (e.g. `static` variable)

I believe with the new behavior of SPIRV-Tools, we should *never* create a temporary variable because of storage class mismatch. (we still do need to create a temporary variable if the param is an rvalue). 
If you're not sure, you can check with @s-perron about what the unoptimized output of DXC should look like.